### PR TITLE
Login page

### LINKED ISSUE #22: User login page
Design and implement a user login page.

### PR DESCRIPTION
**Pull request by VictoriaKGN**
- implemented login page
- integrated UI with backend
- text fields get validated before submission
- userid and token gets globally saved for future requests

image

closes #22

### CODE REVIEW

File: client/src/api/UserAPI.ts [hunk 1]
```diff
@@ -23,3 +30,25 @@ export const signup = async (data: SignupVals): Promise<AxiosResponse> => {
...
+ * @param {string} password - Password
+ * @returns {AxiosResponse}
+ * @throws {AxiosError}
+ */
+export const loginRequest = async (
+  data: LoginVals,
+): Promise<AxiosResponse<LoginResponse>> => {
+  try {
+    const response = await axios.post<LoginResponse>(
+      'http://localhost:7005/users/login',
```
Comment (BarbzCodez) [inline client/src/api/UserAPI.ts:47]: Would this adress work if we run it in docker(?) I think we should be getting the main path `https://localhost:7005` from a function that checks where it is

File: client/src/pages/Login/components/LoginForm.tsx [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        helperText={
+          formik.touched.password && Boolean(formik.errors.password)
+            ? formik.errors.password
+            : ' '
+        }
+      />
+      <Typography
+        variant="body1"
+        align="center"
+        color="error"
```
Comment (BarbzCodez) [inline client/src/pages/Login/components/LoginForm.tsx:114]: would this error color be the deafult error or the error that is on the theme pallete?

File: client/src/api/UserAPI.ts (hunk 1 above)
Comment (VictoriaKGN) [reply client/src/api/UserAPI.ts:47]: yes, this is the exact address that you send your requests to with docker when the server is running

File: client/src/pages/Login/components/LoginForm.tsx (hunk 2 above)
Comment (VictoriaKGN) [reply client/src/pages/Login/components/LoginForm.tsx:114]: this is the error color we set in the global theme, so the colors are consistent throughout the app

**Review by DucharmeEthan — APPROVED** [review]

**Review by BarbzCodez — APPROVED** [review]
Forgot to approve lol

**Pull request merged**